### PR TITLE
add device_vendor to the GpuDevice in JAX runtime

### PR DESCRIPTION
**Pull request by reza-amd**
With this PR, device vendor will be exposed from JAX GpuDevice into Python

@hawkinsp

### CODE REVIEW

File: tensorflow/compiler/xla/pjrt/gpu_device.h [hunk 1]
```diff
@@ -28,7 +28,12 @@ namespace xla {
 class GpuDevice : public PjRtStreamExecutorDevice {
  public:
   GpuDevice(int id, std::unique_ptr<LocalDeviceState> local_device_state,
-            std::string device_kind, int node_id);
+            std::string device_kind, std::string device_vensor, int node_id);
```
Comment (hawkinsp) [inline tensorflow/compiler/xla/pjrt/gpu_device.h:31]: Nit: typo `device_vendor`

**Review by hawkinsp — CHANGES_REQUESTED** [review]
One tiny typo needs fixing.

File: tensorflow/compiler/xla/pjrt/gpu_device.h (hunk 1 above)
Comment (reza-amd) [reply tensorflow/compiler/xla/pjrt/gpu_device.h]: Done.

**Review by hawkinsp — APPROVED** [review]

**Comment by hawkinsp** [comment]
One of our internal presubmits reminded me that you also need to update the Python type stub for pybind code:
https://github.com/tensorflow/tensorflow/blob/0c13207eeda65754532bab5888cc33693fb06834/tensorflow/compiler/xla/python/xla_extension/__init__.pyi#L248
to add `device_vendor`

**Review by hawkinsp — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Comment by hawkinsp** [comment]
Our presubmits are still unhappy, with this error:
```
error: field 'device_vendor_' will be initialized after base 'xla::PjRtStreamExecutorDevice' [-Werror,-Wreorder-ctor]
```

**Review by hawkinsp — APPROVED** [review]

**Pull request merged**